types/meteor/accounts-base.d.ts

```diff
@@ -106,9 +111,9 @@ declare module "meteor/accounts-base" {
 
         function onCreateUser(func: (options: { profile?: {} }, user: Meteor.User) => void): void;
 
-        function findUserByEmail(email: string): Meteor.User | null | undefined;
+        function findUserByEmail(email: string, options?: FieldSpecifier): Meteor.User | null | undefined | any;
 
-        function findUserByUsername(username: string): Meteor.User | null | undefined;
+        function findUserByUsername(username: string, options?: FieldSpecifier): Meteor.User | null | undefined | any;
```

Comment:
As per the [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) in the readme, returning `| any` here isn't very useful, since that's equivalent to not having the union at all. I think this needs to be made more specific to make sense.

Reply:
@ebroder the problem makes any specific type here is possibility to select any field from object which will be returned from User object, which also includes user-defined fields and makes the situation looks a simple blackbox from a types point of view.

Reply:
Small addition: since default fields to be returned now could be configured via `Accounts.config` definition of eg `findUserByEmail` as a function that returns one of `Meteor.User | null | undefined` isn't true since in case any `FieldSpecifier` was passed at startup returned data will not match Meteor.User type